ffe: add page (#6261)
tldr-pages/tldr: pages/common/ffe.md

# ffe

> Extract fields from a flat database file and write to another format.

> > `ffe` needs a text config file to interpret the input and format the output.

Reviewer: This line isn't needed IMO, the fact very command has a config file used, kind of gives it away

Author: Disagree. ‘ffe” can also use a default ~/fferc config with a collection of structures and formats that you commonly use and select from. The point is that it needs _some_ config file and there is no avoiding writing one to describe your data. I will remove -c from the -s and -f examples and mention ~/fferc there to illustrate this mode better.

Reviewer 2: What about this?
```suggestion
> A configuration file is required to interpret the input and format the output.
```

Author: I incorporated this.

> More information: <http://ff-extractor.sourceforge.net/ffe.html>.

- Display all input data using the specified data configuration:

`ffe --configuration={{path/to/config.ffe}} {{path/to/input}}`

- Convert an input file to an output file in a new format:

`ffe --output={{path/to/output}} -c {{path/to/config.ffe}} {{path/to/input}}`

- Select input structure and print format from definitions in `~/.fferc` config file:

> `ffe --structure={{structure}} --print={{format}} {{path/to/input}}`

Reviewer 2: What is "format" in this instance?

Author: Both {{structure}} and {{format}} are defined in the (not shown) configuration file.

- Write only the selected fields:

> `ffe -f "{{FirstName,LastNAme,Age}}" -c {{path/to/config}} {{path/to/input}}`

Reviewer: ```suggestion
`ffe -f "{{FirstName,LastName,Age}}" -c {{path/to/config}} {{path/to/input}}`
```

Author: I’ll fix the case. The field names are samples matching the data not file names.

Reviewer: we use snake case regardless.

Reviewer 3: @CleanMachine1 but we should respect the case used as standard in the target database (e.g. a page for MySQL), I think in this case we don't have one. In the `mongoexport` page we use `field1,field2` so the camelCase used in MongoDB is not affected.

Also, we use snake case but in lower case for placeholders. Wdyt @sbrl ?

Reviewer 2: In this case, we should stick with the existing structure since it looks like it has specific technical meaning. There is a typo here though, which we can fix:

```suggestion
`ffe -f "{{FirstName,LastName,Age}}" -c {{path/to/config}} {{path/to/input}}`
```

If it has already been changed  then there's little point in changing it back, but this is what I would do.

Reviewer 3: @daniel-birket what case do you think is a better fit here?

Author: The sample field names in the --file-list sample were from the author's documentation. I purposely departed from the snake_case used required for tokens in your style guide because they were not tokens but a sample of the format required for the parameter. I have since changed it to Cobra_Case as First_Name,Last_Name as suggested, which will serve as well.

Reviewer 2: In this case, let's match the author's documentation convention. This isn't the same as most other token syntax examples, since it's a specific example and not a descriptive stand-in.

Author: Ok, I'll change it back to the author's sample format.
